### PR TITLE
Added Tarjan's Algorithm.

### LINKED ISSUE #523: [Feature Request]  : Adding Tarjan's Algorithm documentation.
### Is your feature request related to a problem? Please describe.
I'm frustrated when I need to find reliable resources explaining Tarjan's Algorithm for finding strongly connected components, but the project currently lacks detailed documentation on it.

### Describe the solution you'd like
A well-structured documentation page explaining Tarjan's Algorithm, including its use cases, time complexity, pseudocode, and an example implementation. This will help users understand the algorithm and use it effectively in the project.

### Describe alternatives you've considered
Alternative solutions include linking external sources for Tarjan’s Algorithm or adding comments in the code, but these might not provide a thorough explanation tailored to the project.

### Additional context
Add diagrams, step-by-step explanations, and a real-world example where Tarjan's Algorithm is applied for better understanding.

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by AbhijitMotekar99**
### Description
Added Tarjan's Algoritm

Fixes #523 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Tarjan's Algorithm/Tarjan's.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+---
+id: Tarjan's
```
Comment (ajay-dhangar) [inline docs/Tarjan's Algorithm/Tarjan's.md]: replace `id: Tarjan's`1 with `id: tarjan`

File: docs/Tarjan's Algorithm/Tarjan's.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+title: Introduction to Tarjan's Algorithm
```
Comment (ajay-dhangar) [inline docs/Tarjan's Algorithm/Tarjan's.md]: replace `title: Introduction to Tarjan's Algorithm` with `title: "Introduction to Tarjan's Algorithm"`

File: docs/Tarjan's Algorithm/Tarjan's.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+sidebar_label: Tarjan's Algorithm
```
Comment (ajay-dhangar) [inline docs/Tarjan's Algorithm/Tarjan's.md]: replace `sidebar_label: Tarjan's Algorithm` with `sidebar_label: "Tarjan's Algorithm"`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on suggestions

**Comment by AbhijitMotekar99** [comment]
Hey @ajay-dhangar 
I done all changes.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**